internal/pkg/agent/application/monitoring/v1_monitor.go

```diff
@@ -714,6 +714,69 @@ func (b *BeatsMonitor) injectMetricsInput(cfg map[string]interface{}, componentI
 				},
 			},
 		})
+
+		if strings.EqualFold(name, "filebeat") {
+			fbDataStreamName := "filebeat_input"
+			streams = append(streams, map[string]interface{}{
+				idKey: "metrics-monitoring-" + name + "-1",
+				"data_stream": map[string]interface{}{
+					"type":      "metrics",
+					"dataset":   fmt.Sprintf("elastic_agent.%s", fbDataStreamName),
+					"namespace": monitoringNamespace,
+				},
+				"metricsets":    []interface{}{"json"},
+				"hosts":         endpoints,
+				"path":          "/inputs/",
+				"namespace":     fbDataStreamName,
+				"json.is_array": true,
+				"period":        "10s",
+				"index":         fmt.Sprintf("metrics-elastic_agent.%s-%s", fbDataStreamName, monitoringNamespace),
```

Comment:
make sure that we have permissions to write to this index. 
we have a limited output permissions (part of the incoming policy `output_permissions`)

Reply:
The permission to auto generate `filebeat_input` indices is in this merged PR: https://github.com/elastic/kibana/pull/149974